src/System.Reflection/tests/FieldInfoTests.cs

```diff
@@ -0,0 +1,526 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+#pragma warning disable 0414
+
+namespace System.Reflection.Tests
+{
+    public class FieldInfoTests
+    {
+        [Theory]
+        [InlineData(nameof(FieldInfoTests.ConstIntField), 222)]
+        [InlineData(nameof(FieldInfoTests.ConstStringField), "new value")]
+        [InlineData(nameof(FieldInfoTests.ConstCharField), 'A')]
+        [InlineData(nameof(FieldInfoTests.ConstBoolField), false)]
+        [InlineData(nameof(FieldInfoTests.ConstFloatField), 4.56f)]
+        [InlineData(nameof(FieldInfoTests.ConstDoubleField), double.MaxValue)]
+        [InlineData(nameof(FieldInfoTests.ConstInt64Field), long.MaxValue)]
+        [InlineData(nameof(FieldInfoTests.ConstByteField), byte.MaxValue)]
+        public void SetValue_ConstantField_ThrowsFieldAccessException(string field, object value)
+        {
+            FieldInfo fieldInfo = GetField(typeof(FieldInfoTests), field);
```

Comment:
I see; it's not necessary because we wouldn't get back the FieldInfo if the name was different.